events/api.py

```diff
@@ -529,6 +529,23 @@ def create(self, validated_data):
                 raise error
         return instance
 
+    def create_keyword(self, validated_data):
+        if 'data_source' not in validated_data:
```

Comment:
Feels kind of icky that a `create`... function is mutating data that's passed in.